django/http/multipartparser.py

```diff
@@ -682,21 +683,36 @@ def parse_boundary_stream(stream, max_header_size):
     """
     Parse one and exactly one stream that encapsulates a boundary.
     """
-    # Stream at beginning of header, look for end of header
-    # and parse it if found. The header must fit within one
-    # chunk.
-    chunk = stream.read(max_header_size)
 
-    # 'find' returns the top of these four bytes, so we'll
-    # need to munch them later to prevent them from polluting
-    # the payload.
-    header_end = chunk.find(b"\r\n\r\n")
+    # Start with checking for the end of headers in the first kilobyte and if not found
+    # extend the search to double the size (up to the `max_header_size`).
+    headers_chunk_size = 1024
+    while True:
+        if headers_chunk_size > max_header_size:
```

Comment:
Is it worth doing to search in chunks? For now it seems unnecessary :thinking: `find()` works pretty fast and we don't need to be prepared for very big values in `MAX_TOTAL_HEADER_SIZE`. I'd revert changes to the `parse_boundary_stream()`.

Reply:
My intention was to keep the default lookup chunk size as low as it was before which should capture the absolute most of the cases and would avoid extending the necessary memory footprint.

Reply:
I'm not sure we need this 🤷 . MAX_TOTAL_HEADER_SIZE cannot be change accidentally, so folks who need to do this can optimize it's value for their specific use case.

Reply:
This was to avoid 'accidental' DoS just by changing limit. So the value proposed is that it would keep the runtime characteristics with regards to performance and memory consumption for the most cases even if the limit is configured higher and therefore it would only change these for when the exceptional parts appear.